src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,254 @@
+/**
+Copyright 2020 JasmineGraph Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+**/
+
+#include "NodeManager.h"
+
+#include <sys/stat.h>
+
+#include "../../util/logger/Logger.h"
+#include "NodeBlock.h"  // To setup node DB
+#include "PropertyLink.h"
+#include "RelationBlock.h"
+
+Logger node_manager_logger;
+
+NodeManager::NodeManager(std::string mode) {
+    std::ios_base::openmode openMode = std::ios::trunc;  // default is Trunc mode which overrides the entire file
+    if (mode == "app") {
+        openMode = std::ios::app;  // if app, open in append mode
+        this->nodeIndex = readNodeIndex();
+    }
+    NodeBlock::nodesDB =
+        new std::fstream(NodeManager::NODE_DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    PropertyLink::propertiesDB =
+        new std::fstream(PropertyLink::DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    RelationBlock::relationsDB =
+        new std::fstream(RelationBlock::DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    // TODO (tmkasun): set PropertyLink nextPropertyIndex after validating by modulus check from file number of bytes
+
+    if (dbSize(NodeManager::NODE_DB_PATH) % NodeBlock::BLOCK_SIZE != 0) {
+        node_manager_logger.error("Node DB size does not comply to node block size Path = " +
+                                  NodeManager::NODE_DB_PATH);
+    }
+}
+
+std::unordered_map<std::string, unsigned int> NodeManager::readNodeIndex() {
+    std::ifstream index_db(this->index_db_loc, std::ios::app | std::ios::binary);
+    std::unordered_map<std::string, unsigned int> _nodeIndex;  // temporary node index data holder
+
+    if (index_db.is_open()) {
+        int iSize = dbSize(this->index_db_loc);
+        unsigned long dataWidth = NodeManager::INDEX_KEY_SIZE + sizeof(unsigned int);
+        if (iSize % dataWidth != 0) {
+            node_manager_logger.error("Index DB size does not comply to index block size Path = " +
+                                      this->index_db_loc);
+            throw std::runtime_error("Node index DB in " + this->index_db_loc + " is corrupted!");
+        }
+
+        char nodeIDC[NodeManager::INDEX_KEY_SIZE];
+        unsigned int nodeIndexId;
+        for (size_t i = 0; i < iSize / dataWidth; i++) {
+            nodeIDC[NodeManager::INDEX_KEY_SIZE] = {0};  // Fill with null chars before puting data
```

Comment:
Change `puting` to `putting`

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done